lib/combine_pdf/pdf_public.rb

```diff
@@ -187,7 +187,7 @@ def to_pdf(options = {})
       xref = []
       indirect_object_count = 1 # the first object is the null object
       # write head (version and binanry-code)
-      out << "%PDF-#{@version}\n%\xFF\xFF\xFF\xFF\xFF\x00\x00\x00\x00".force_encoding(Encoding::ASCII_8BIT)
+      out << "%PDF-#{@version}\n%\xFF\xFF\xFF\xFF\xFF\x00\x00\x00\x00".dup.force_encoding(Encoding::ASCII_8BIT)
```

Comment:
If the interpreter requires `dup` it's a little odd... Maybe there's a way to indicate that a String literal is ASCII without creating an additional object (see: https://stackoverflow.com/a/15843685/4025095 )?

After all, the object's lifetime os very short (only created to be inserted into the existing String).